R/loadFromMetaphlan.R

```diff
@@ -149,20 +149,27 @@ loadFromMetaphlan <- function(file, sample_meta = NULL, phy_tree = NULL, ...){
 }
 
 # Check that metaphlan file contains correct information
-.check_metaphlan <- function(data){
-    # Get rowdata columns
-    rowdata_columns <- data[ , 1:2]
-    # Get columns that go to assay
-    assay_columns <- data[ , 3:ncol(data)]
+.check_metaphlan <- function(x){
+    # Getting column indices of character columns
+    col_idx <- sapply(x, is.character)
+    # Get rowdata column indices
+    col_rowdata_idx <- col_idx[col_idx==TRUE]
+    # Getting column indices of numeric columns
+    col_idx <- sapply(x[,!col_idx], is.numeric)
+    # Get assay column indices
+    col_assay_idx <- col_idx[col_idx==TRUE]
```

Comment:
see above. Have you ran this code line by line to check the output?

Reply:
yes,  I tried line by line before pushing changes. I will switch to `which`

Reply:
col_idx[col_idx==TRUE] will return logical vector whose length is sum(col_idx==TRUE) and all elements are TRUE.